### ISSUE TITLE
build error: "DSO missing from command line" (boost 1.67)

### ISSUE THREAD

**Issue by sebix** [aguinet/usbtop#16]
I get this build error with gcc8
```
[   54s] [100%] Linking CXX executable usbtop
[   54s] cd /home/abuild/rpmbuild/BUILD/usbtop-release-0.2/build/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/usbtop.dir/link.txt --verbose=1
[   54s] /usr/bin/c++  -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -DNDEBUG -O2 -g -DNDEBUG  -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now -rdynamic CMakeFiles/usbtop.dir/buses.cpp.o CMakeFiles/usbtop.dir/console_output.cpp.o CMakeFiles/usbtop.dir/main.cpp.o CMakeFiles/usbtop.dir/pcap_compat.cpp.o CMakeFiles/usbtop.dir/should_stop.cpp.o CMakeFiles/usbtop.dir/tools.cpp.o CMakeFiles/usbtop.dir/usb_bus.cpp.o CMakeFiles/usbtop.dir/usb_device.cpp.o CMakeFiles/usbtop.dir/usb_stats.cpp.o  -o usbtop -lpcap -lboost_thread -lboost_system -lboost_chrono -lboost_date_time 
[   54s] /usr/lib64/gcc/x86_64-suse-linux/8/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/usbtop.dir/main.cpp.o: undefined reference to symbol 'pthread_condattr_setclock@@GLIBC_2.3.3'
[   54s] /lib64/libpthread.so.0: error adding symbols: DSO missing from command line
[   54s] collect2: error: ld returned 1 exit status
[   54s] make[2]: *** [src/CMakeFiles/usbtop.dir/build.make:212: src/usbtop] Error 1
[   54s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/usbtop-release-0.2/build'
[   54s] make[1]: *** [CMakeFiles/Makefile2:89: src/CMakeFiles/usbtop.dir/all] Error 2
[   54s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/usbtop-release-0.2/build'
[   54s] make: *** [Makefile:133: all] Error 2
```

According to a [Stackoverflow Thread](https://stackoverflow.com/questions/19901934/strange-linking-error-dso-missing-from-command-line):
> the linking is dependent on the order of modules. Symbols are first requested, and then linked in from a library that has them. So you have to specify modules that use libraries first, and libraries after them.

I don't know what dependency upgrade actually caused the fail.

**Comment by sebix** [comment]
This patch is now used in openSUSE for gcc8: https://build.opensuse.org/package/view_file/utilities/usbtop/usbtop-fix-build-boost.patch?expand=1

**Comment by trofi** [comment]
I think it's a cmake bug. Likely fixed in cmake-3.13.0: https://gitlab.kitware.com/cmake/cmake/commit/bd831ed0948a1e99f573f0056f2bee5d3b21009e

**Comment by aguinet** [comment]
I just released usbtop 1.0 which removes the binary dependency on boost. You may try that one!

**Comment by aguinet** [comment]
Please re-open if the problem is still there!
